RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -112,6 +113,40 @@ HcalNoiseInfoProducer::HcalNoiseInfoProducer(const edm::ParameterSet& iConfig) :
     edm::LogWarning("HCalNoiseInfoProducer") << " forcing fillRecHits to be true if fillDigis is true.\n";
   }
 
+  // get the fiber configuration vectors
+  std::vector<int> tmpLaserMonDetTypeList = iConfig.getParameter<std::vector<int> >("laserMonDetTypeList");
+  std::vector<int> tmpLaserMonIPhiList = iConfig.getParameter<std::vector<int> >("laserMonIPhiList");
+  std::vector<int> tmpLaserMonIEtaList = iConfig.getParameter<std::vector<int> >("laserMonIEtaList");
+
+  // the transfer of data from python to c seems to have issues
+  // this can be fixed by explicitly filling the vector
```

Comment:
???

Reply:
There seems to be some memory issue here that I wasn't able to resolve otherwise.  If I simply assign the member variables directly with getParameter (and eliminate the code that push_backs the vectors) and then print the values, it looks OK.  However, in the produce() function I find that the vectors are empty (ie ( laserMonIEtaList_.size() == 0 ) evaluates to True and the same for the other two vectors).  Have you seen this before?

Reply:
What exactly was the code you were trying to use previously? That would help diagnose what was happening.

Reply:
I've made a simple patch file that you can apply as below, recompile, and do cmsRun reco_RAW2DIGI_RECO.py (from my example setup above) to reproduce my printout.

patch RecoMET/METProducers/src/HcalNoiseInfoProducer.cc  -i /afs/cern.ch/user/j/jkunkle/public/HcalLaserCodeTest/printtest.patch  -o HcalNoiseInfoProducer.cc
mv HcalNoiseInfoProducer.cc RecoMET/METProducers/src/
scram b
cmsRun reco_RAW2DIGI_RECO.py 

You should see in the stdout

---

Vector sizes in constructor : 
laserMonIPhiList_.size() = 8
laserMonIEtaList_.size() = 8
laserMonDetTypeList_.size() = 8
---
Vector sizes in produce : 
laserMonIPhiList_.size() = 0
laserMonIEtaList_.size() = 0
laserMonDetTypeList_.size() = 0

Reply:
The problem is here in the constructor:
```C++
  std::vector<int> laserMonDetTypeList_= iConfig.getParameter<std::vector<int> >("laserMonDetTypeList");
  std::vector<int> laserMonIPhiList_   = iConfig.getParameter<std::vector<int> >("laserMonIPhiList");
  std::vector<int> laserMonIEtaList_   = iConfig.getParameter<std::vector<int> >("laserMonIEtaList");
```

you redefined the variable names to be function local variables instead of member variables. The fix is to remove the type declaration:

```C++
  laserMonDetTypeList_= iConfig.getParameter<std::vector<int> >("laserMonDetTypeList");
  laserMonIPhiList_   = iConfig.getParameter<std::vector<int> >("laserMonIPhiList");
  laserMonIEtaList_   = iConfig.getParameter<std::vector<int> >("laserMonIEtaList");
```

Reply:
ah my mistake, I should have seen that.  Thanks